### ISSUE TITLE
NoSuchMethodError on Java 10+

### ISSUE THREAD

**Issue by stelfrich** [fiji/fiji#211]
```
java.lang.NoSuchMethodError: java.awt.MenuBar.getPeer()Ljava/awt/peer/MenuComponentPeer;
	at fiji.IJ_Alt_Key_Listener.getX11Opener(IJ_Alt_Key_Listener.java:98)
	at fiji.IJ_Alt_Key_Listener.getOpener(IJ_Alt_Key_Listener.java:87)
	at fiji.IJ_Alt_Key_Listener.<init>(IJ_Alt_Key_Listener.java:18)
	at fiji.DefaultFijiService.actuallyInitialize(DefaultFijiService.java:55)
	at fiji.DefaultFijiService.onEvent(DefaultFijiService.java:61)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.scijava.event.DefaultEventService$ProxySubscriber.onEvent(DefaultEventService.java:301)
	at org.scijava.event.DefaultEventService$ProxySubscriber.onEvent(DefaultEventService.java:275)
	at org.bushe.swing.event.ThreadSafeEventService.publish(ThreadSafeEventService.java:971)
	at org.scijava.event.DefaultEventBus.access$201(DefaultEventBus.java:57)
	at org.scijava.event.DefaultEventBus$2.run(DefaultEventBus.java:217)
	at org.scijava.thread.DefaultThreadService$2.run(DefaultThreadService.java:221)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:514)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1135)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:844)
```

**Comment by turekg** [comment]
What OS please?

**Comment by stelfrich** [comment]
> Also a couple of additional notes (probably known issues, but worth noting):

Thanks for reporting @turekg. There are issues of [`imagej-launcher`](https://github.com/imagej/imagej-launcher) for both of those (don't have the numbers at hand at the moment). The first one has already been fixed in the latest `SNAPSHOT` version. The second one will be a bit more tricky..

**Comment by ctrueden** [comment]
This issue was already fixed with 8f2736a5f8c85d0d572a0203d80d86c7e3f14589.

@stelfrich Maybe the issue is that `fiji-compat` is also still present in the `jars` folder? If so, this issue will be fully resolved by the switch to the new v2 Updater. Unfortunately, I think we cannot mark fiji-compat obsolete as long as the Java-8 update site continues to be in production.

**Comment by pettaa123** [comment]
Hi, I installed recent 64 eclipse and 64 JDK 12.0.1 on my Windows 10 64 machine. I cloned recent fiji master and did the import via maven. I get the following errors:

`Description	Resource	Path	Location	Type
The method getPeer() is undefined for the type MenuBar	IJ_Alt_Key_Listener.java	/fiji/src/main/java/sc/fiji/compat	line 97	Java Problem`

`Description	Resource	Path	Location	Type
The method getPeer() is undefined for the type MenuBar	IJ_Alt_Key_Listener.java	/fiji/src/main/java/sc/fiji/compat	line 113	Java Problem`

I would like to learn how to debug the stitiching plugin. Does this error prevent me from doing that?

**Comment by ctrueden** [comment]
A further related fix: fiji/fiji@15f288f4af6a929c7b16ec4373b68be01d8b64f4. Again, removing `jar/fiji-compat.jar` is also required to fully fix this.

@pettaa123 Sorry to ignore you for so long.
> I would like to learn how to debug the stitiching plugin. Does this error prevent me from doing that?

Right now, for all development, I would still encourage you to stay with Java 8, not Java 12. There may be issues with newer Javas which make your life more difficult, yes.

**Comment by ctrueden** [comment]
I'm going to close this, because since the above was written, we were able to effectively remove fiji-compat from the `jars` folder by replacing it with an empty stub. Until the Java-8 update site finally splits again and dies, we are using this trick to remove any JARs present on upstream (ImageJ/Fiji) update sites.
